### PR TITLE
bpo-36548: Improve the repr of re flags.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue36548

### CODE REVIEW

File: Lib/re.py [hunk 1]
```diff
@@ -141,24 +141,40 @@
...
-    DOTALL = sre_compile.SRE_FLAG_DOTALL # make dot match newline
-    VERBOSE = sre_compile.SRE_FLAG_VERBOSE # ignore whitespace and comments
-    A = ASCII
-    I = IGNORECASE
-    L = LOCALE
-    U = UNICODE
-    M = MULTILINE
-    S = DOTALL
-    X = VERBOSE
+    ASCII = A = sre_compile.SRE_FLAG_ASCII # assume ascii "locale"
```
Comment (SylvainDe) [inline Lib/re.py:144]: Not related to the issue you are solving (but as you're updating this code anyway):
 - the alignment may be more aesthetically pleasing if the one latter variable was the first here
 - PEP8 recommends using at least 2 whitespace before the inline comment

File: Lib/re.py [hunk 2]
```diff
@@ -141,24 +141,40 @@
...
+        if value:
+            members.append(hex(value))
+        res = '|'.join(members)
+        if negative:
+            if len(members) > 1:
+                res = f'~({res})'
+            else:
+                res = f'~{res}'
+        return res
+    __str__ = object.__str__
```
Comment (SylvainDe) [inline Lib/re.py:176]: As you're adding this line, would it be worth adding a few test cases for it ?

File: Lib/re.py (hunk 1 above)
Comment (serhiy-storchaka) [reply Lib/re.py:144]: In case of aliases the canonical name is the one that is defined first. We want to use long names.

Comment (serhiy-storchaka) [reply Lib/re.py:144]: Some lines are already too long, and using two spaces will make them even longer.

**Pull request merged**